### ISSUE TITLE
Feature Request: 

### ISSUE THREAD

**Issue by cjohn001** [williamjuan027/nativescript-movies-demo#4]
@williamjuan027 : I am currently trying to follow your animation tutorial at

https://indepth.dev/posts/1285/in-depth-guide-into-animations-in-angular

I am trying to animate a dialog overlay in a transparent grid which prevents the user from interacting with the app when the dialog is open.

<GridLayout columns="*" *ngIf="!_dialogVisible" @container>
 <dialog @inOutAnimation></dialog>
</GridLayout>

according to your tutorial I understood that I need a trigger on the parent (GridLayout) which calls animateChild.

animations: [
		trigger('container', [transition(':enter, :leave', [query('@*', animateChild() ) ])]),
		trigger('inOutAnimation', [
			transition(':enter', [
				style({ transform: 'translateY(600)' }),
				animate('1350ms ease-in-out', style({ transform: 'translateY(10)' }))
			]),
			transition(':leave', [
				style({ transform: 'translateY(10)' }),
				animate('1350ms ease-in-out', style({ transform: 'translateY(600)' }))
			])
		])
	]

Unfortunately, I cannot get the out animation to work. Seems like the ngIf on the GridLayout is not awaiting the child transitions. Would be great if you could show how this can be done with nativescript. Thx!

**Comment by williamjuan027** [comment]
Hey @cjohn001 

That is a great question. I've only tested that on a web Angular project, I haven't tried it on NativeScript yet. Let me try that on this project and get back to you.

As an alternative, you could use a different method to block user interactions when a popup opened this way is added. You could structure your html this way:

```html
<GridLayout rows="*" columns="*">

   <!-- your shade layer (instead of a parent, this will be a layer behind the dialog) -->
   <ContentView *ngIf="! _dialogVisible" height="100%" width="100%" (tap)="noop()"></ContentView>

   <!-- your actual dialog here -->
   
   
</GridLayout>
```

and on your typescript file:

```typescript
noop(): void {
  // nothing here - this is just to catch the tap events so it doesn't go through to the layers behind it
}
```

**Comment by cjohn001** [comment]
@williamjuan027 : Thank you very much for taking the time to look into it, and also for sharing your idea with a modified code structure. It would work like you sketched it if I would only like to animate the dialog. Attached video shows what I am trying to program. The current animations are done with the nativescript animation method. You can see that the background page is scaled down a little to give it perspective. This is happening with the dialog animation at the same time. Hence, I assume I would required a more complex animation logic with angular which can trigger. child animations with the background view scaling. Would be very interested how this could be done with angular animation framework. 

https://user-images.githubusercontent.com/43253483/134067492-a645695b-484e-4574-a3b9-062d7a143e36.mov

**Comment by williamjuan027** [comment]
That's a really cool animation. I've never done that with Angular animations on a NativeScript app, I've used NativeScript's animation APIs for that. Let me try a similar animation in this app. Thanks for bringing this up.

**Comment by williamjuan027** [comment]
@cjohn001 I'm debugging this and there seems to be a problem with the leave animation. Are you having any issues with just animating an ngIf's exit on a single element. Perhaps something like this:

```html
<StackLayout *ngIf="isDisplayed" @fade></StackLayout>
```

```typescript
trigger("fade", [
  transition(":enter", [
    style({ opacity: 0 }),
    animate("300ms 100ms ease-in", style({ opacity: 1 })),
  ]),
  transition(":leave", [
    style({ opacity: 1 }),
    animate("300ms 100ms ease-in", style({ opacity: 0 })),
  ]),
])
```

**Comment by cjohn001** [comment]
@williamjuan027: Yes I had an issue there also. It seems like the object is directly removed from DOM, no animation happening.

**Comment by williamjuan027** [comment]
@cjohn001 There could be a regression in core that affected the `ngIf` leave animation. `animateChild` seems to work, but leave animation, regardless of within the `animateChild` or not doesn't wait for the animation to end before removing the element from the DOM. I'm going to test this against an older version to verify that it was working in prior versions.

The second observation you mentioned seems to be related to `NgZone`. If the variable used in the `ngIf` is coming from an external source that runs outside of `NgZone`, it's possible that it doesn't update the DOM correctly. You could maybe wrap wherever the variable gets updated inside the following code:

```typescript
import { NgZone } from '@angular/core';

constructor(private ngZone: NgZone) {}

this.ngZone.run(() => {
  // put the code where you update the variable here
})
```

**Comment by cjohn001** [comment]
@williamjuan027 : Thanks for the update.  So I assume you than also have not gotten the animation running? Seems like it is to early than to jump to angular animations with nativescript.  Could you provide an issue on the @nativescript/angular repo. The plugin recently got an update. Maybe, there is a chance that someone of the maintainers looks at it.

**Comment by cjohn001** [comment]
Hello @williamjuan027, I was wondering if you have found an issue to the topic. Would be great if one could dynamically load dialogs when they are opened via the animation. Currently I have lots of dialogs in certain pages which makes loading of the initial page very slow.

**Comment by williamjuan027** [comment]
Hey @cjohn001
You could use `RootLayout`. The views aren't loaded until you they are rendered and it gets destroyed as they are closed. Do you think that would work for your use case?

**Comment by cjohn001** [comment]
@williamjuan027 : Thanks for sharing the idea with the RootLayout. I thought about it as a last resort indeed. However, in the meantime I have found a better solution. I have not worked with the RootLayout yet, but it seems I than would need to load the dialog from code. Hence, I could not use Template bindings to create the dialogs, which is the preferred way for me. For example I have:

```
<ns-edit-weekday-dialog #idWeekdayDialog *ngIf="_renderWeekdayDialog" [_selectedIdx]="_selectedWeekdayIdx"
    (valueChanged)="onWeekdayChanged($event)" [_parentViewToOverlay]="idScrollView">
 </ns-edit-weekday-dialog>
```

```
public onChangeWeekday() {
		this._renderWeekdayDialog = true;
		this.cdr.detectChanges();
		setTimeout(() => {
			this._weekdayButtonTextField.nativeElement._goToVisualState('');
			this._weekdayDialog.showDialog(true);
		});
	}

```

Once I had the detectChanges() call after the this._renderWeekdayDialog = true things started working. 

Nevertheless, it would be great if you could provide an issue report regarding your findings (I think a recursion in core you had found?) in the ns-angular repo. It would be great if we could use the Angular animation framework for everything, and maybe one of the maintainers gets this thing fixed.

**Comment by williamjuan027** [comment]
Issue with `*ngIf` not triggering `leave` animations - https://github.com/NativeScript/angular/issues/52
